### ISSUE TITLE
RDAirplay, RDCartSlots (qt5 beta test) seems broken after commits August/31/2021

### ISSUE THREAD

**Issue by alexolivan** [ElvishArtisan/rivendell#719]
Hi. Today I've found new version of Rivendell since I'm using nightly builds.

This is the first package I deploy after successfully rebuilding packages with hpklinux instead of libhpi (still I have pending to perform a fully clean rebuild).
Update was somehow troublesome since I had to remove Audioscience libhpi and install hpklinux instead, but once done, it installed OK and I performed a ```rddbmgr --modify``` for my first time

Now RDAirplay seems broken:

On rdairplay invoking console  (It loads last logs automatically) right after run errors like this show up:
```
QMYSQLResult::data: column 10 out of range
for query: select `NAME`,`DESCRIPTION`,`DEFAULT_LOW_CART`,`DEFAULT_HIGH_CART`,`ENFORCE_CART_RANGE`,`DEFAULT_CART_TYPE`,`NOTIFY_EMAIL_ADDRESS`,`REPORT_TFC`,`REPORT_MUS`,`COLOR` from `GROUPS`  where ((`GROUPS`.`NAME`='ALL')||(`GROUPS`.`NAME`='BEDS')||(`GROUPS`.`NAME`='IDENTS')||(`GROUPS`.`NAME`='LEGAL')||(`GROUPS`.`NAME`='LOGOS')||(`GROUPS`.`NAME`='MACROS')||(`GROUPS`.`NAME`='MUSIC')||(`GROUPS`.`NAME`='RECORDS')||(`GROUPS`.`NAME`='STINGS')||(`GROUPS`.`NAME`='TEMP')||(`GROUPS`.`NAME`='TEST')||(`GROUPS`.`NAME`='TRAFFIC')||(`GROUPS`.`NAME`='VOICETRACK')) order by `NAME`
```

When trying to resume the playout on the MainLog, even if re-doing the playout with rdlogmanager, the following error is shown in syslog for every single event on the log and the following next/log as it logchains them (lots of lines like the following):
```
Sep  1 12:34:10 debian rdairplay: log engine: finished event: Line: 317  Cart: 10273  Cut: 1 Card: 0  Stream: -1  Port: 0
```
Then, afterwards, lots like the following:
```
Sep  1 12:34:14 debian rdairplay: invalid SQL or failed DB connection[Column 'EVENT_DATETIME' cannot be null QMYSQL: Unable to execute query]: insert into `ELR_LINES` set `SERVICE_NAME`='Musica',`LENGTH`=0,`LOG_NAME`='Musica-09022021',`LOG_ID`=2,`CART_NUMBER`=10262,`STATION_NAME`='debian',`EVENT_DATETIME`=NULL,`EVENT_TYPE`=3,`EVENT_SOURCE`=2,`EXT_START_TIME`="00:08:00",`EXT_LENGTH`=-1,`EXT_DATA`='',`EXT_EVENT_ID`='',`EXT_ANNC_TYPE`='',`PLAY_SOURCE`=1,`CUT_NUMBER`=1,`EXT_CART_NAME`='',`TITLE`='Figli di Pitagora (extended)',`ARTIST`='Gabry Ponte feat. Little Tony',`SCHEDULED_TIME`="00:08:00",`ISRC`='',`PUBLISHER`='',`COMPOSER`='',`USAGE_CODE`=0,`START_SOURCE`=2,`ONAIR_FLAG`='N',`ALBUM`='Figli di Pitagora',`LABEL`='Universal',`USER_DEFINED`='Imported from Dropbox Music-Electronic-Dance',`CONDUCTOR`='',`SONG_ID`='',`DESCRIPTION`='Cut 001',`OUTCUE`='',`ISCI`=''
```

When loading a cart on the first button of an empty/fresh MainLog, and trying to play it, RDAirplay does nothing, and the cart appears inmediatly greyed as alreeady played. In console this happens:
```
DB connection re-established
invalid SQL or failed DB connection[Column 'EVENT_DATETIME' cannot be null QMYSQL: Unable to execute query]: insert into `ELR_LINES` set `SERVICE_NAME`='Musica',`LENGTH`=0,`LOG_NAME`='Musica-09022021',`LOG_ID`=362,`CART_NUMBER`=10005,`STATION_NAME`='debian',`EVENT_DATETIME`=NULL,`EVENT_TYPE`=3,`EVENT_SOURCE`=0,`EXT_START_TIME`=NULL,`EXT_LENGTH`=-1,`EXT_DATA`='',`EXT_EVENT_ID`='',`EXT_ANNC_TYPE`='',`PLAY_SOURCE`=1,`CUT_NUMBER`=1,`EXT_CART_NAME`='',`TITLE`='Medusa',`ARTIST`='Airbase',`SCHEDULED_TIME`=NULL,`ISRC`='',`PUBLISHER`='',`COMPOSER`='',`USAGE_CODE`=0,`START_SOURCE`=1,`ONAIR_FLAG`='N',`ALBUM`='One Tear Away',`LABEL`='Intuition Recordings',`USER_DEFINED`='Imported from Dropbox Music-Electronic-Trance',`CONDUCTOR`='',`SONG_ID`='',`DESCRIPTION`='Cut 001',`OUTCUE`='',`ISCI`=''
```

I don't think such SQL errors could be related to my new build against hpklinux/hpklinux-dev instead of libhpi-dev
Maybe I'm missing some step necessary with rddbmgr?

UPDATE 1
Performed an ```rddbmgr --check`` with no changes and no apparent errors or problems on the DB. The problem is not fixed, so it seems somethings wrong.
```
root@debian:~# rddbmgr --check
Checking DB/table attributes...
done.

Checking voice tracks...
done.

Checking for stale cart reservations...
done.

Checking schedule code rules...
done.

Checking for orphaned carts...
done.

Checking for orphaned cuts...
done.

Checking cart->cuts counters...
done.

Checking for orphaned audio...
done.

Validating audio lengths (this may take some time)...
done.
```

UPDATE 2
Something very similiar happens with RDCartSlots. Console shows SQL error right from the start (The column 10 out of range error)

**Comment by ElvishArtisan** [comment]
Confirmed here.

**Comment by ElvishArtisan** [comment]
Fixed in 0b58b4f. Please test!

**Comment by alexolivan** [comment]
Confirm its fixed.
Closing
